### PR TITLE
chore(storage): remove `t` prefix in fullkey

### PR DESCRIPTION
**Pull request by Gun9niR**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

The `t` prefix in a `full key` is used to identify whether it contains a table ID. However, only unit tests will have full keys that do not start with a table ID. Therefore, we can make all tests use prefixed keys and then remove the `t` prefix.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/storage/src/hummock/compactor/compactor_runner.rs [hunk 1]
```diff
@@ -85,6 +85,7 @@ impl CompactorRunner {
         filter_key_extractor: Arc<FilterKeyExtractorImpl>,
         task_progress: Arc<TaskProgress>,
     ) -> HummockResult<CompactOutput> {
+        println!("compactor compact: {:?}", self.compact_task.input_ssts);
```
Comment (Li0k) [inline src/storage/src/hummock/compactor/compactor_runner.rs:88]: remove this

File: src/storage/src/hummock/compactor/compactor_runner.rs [hunk 2]
```diff
@@ -95,6 +96,7 @@ impl CompactorRunner {
                 Some(task_progress),
             )
             .await?;
+        println!("{:?}", ssts);
```
Comment (Li0k) [inline src/storage/src/hummock/compactor/compactor_runner.rs:99]: println!

File: src/storage/src/hummock/test_utils.rs [hunk 3]
```diff
@@ -219,9 +219,16 @@ pub async fn gen_test_sstable(
     gen_test_sstable_inner(opts, sst_id, kv_iter, sstable_store, CachePolicy::NotFill).await
 }
 
-/// The key (with epoch 0) of an index in the test table
+pub fn prefixed_key<T: AsRef<[u8]>>(key: T) -> Bytes {
+    let mut buf = Vec::new();
+    buf.put_u32(0);
```
Comment (Li0k) [inline src/storage/src/hummock/test_utils.rs:224]: add some comments for  `prefix` 0 ?

**Review by Li0k — APPROVED** [review]
Rest LGTM, thanks

**Review by skyzh — APPROVED** [review]
Generally LGTM from my side. I remembered there're also some place that we directly take 5 elements out of the full key somewhere (in keyspace?). May need to fix integration tests afterwards.

(Also I feel like I'll have a lot of conflicts to resolve when upgrading toolchains 😇)

**Comment by skyzh** [comment]
... just search for magic number 5 and see if there're more places to fix

File: src/storage/hummock_test/src/test_utils.rs [hunk 4]
```diff
@@ -206,3 +207,11 @@ pub async fn prepare_local_version_manager_new(
 
     (local_version_manager, event_tx, event_rx)
 }
+
+/// Prefix the `key` with table id 0.
+pub fn prefixed_key<T: AsRef<[u8]>>(key: T) -> Bytes {
+    let mut buf = Vec::new();
+    buf.put_u32(0);
```
Comment (wenym1) [inline src/storage/hummock_test/src/test_utils.rs:214]: We may use a test table id instead of 0.

**Review by wenym1 — COMMENTED** [review]
Can we try the following steps to ensure that all unit tests have been explicitly added a table id prefix?
* add a `t` prefix in `prefixed_key`.
* check the `t` prefix in `get_table_id`

**Comment by Gun9niR** [comment]
> Can we try the following steps to ensure that all unit tests have been explicitly added a table id prefix?
> 
> * add a `t` prefix in `prefixed_key`.
> * check the `t` prefix in `get_table_id`

I added the `t` in a [previous commit](https://github.com/risingwavelabs/risingwave/pull/5975/commits/8ffb445a52c7c3979ddd89d5133fe9c3efbbe599), then removed it after all tests have passed.

**Review by wenym1 — APPROVED** [review]

File: src/storage/hummock_test/src/test_utils.rs (hunk 4 above)
Comment (Gun9niR) [reply src/storage/hummock_test/src/test_utils.rs:214]: What should the test table id be? 🤔

File: src/storage/src/hummock/test_utils.rs (hunk 3 above)
Comment (Gun9niR) [reply src/storage/src/hummock/test_utils.rs:224]: I just chose a random number. Anything to watch out about this dummy table id?

File: src/storage/hummock_test/src/test_utils.rs (hunk 4 above)
Comment (wenym1) [reply src/storage/hummock_test/src/test_utils.rs:214]: Something other than 0. `TEST_TABLE_ID = 233u32` may be fine.

Currently we use `TableId::default()` a lot in our unit tests and have many hacks `table_id == 0` in our code to pass the unit tests. Maybe we can remove these hack and pass a more meaningful table id.

File: src/storage/src/hummock/test_utils.rs (hunk 3 above)
Comment (Li0k) [reply src/storage/src/hummock/test_utils.rs:224]: `TableId::default()` , the default value of table_id is 0

**Pull request merged**